modules/swagger-codegen/src/main/resources/swift4/CodableHelper.mustache

```diff
@@ -11,14 +11,20 @@ public typealias EncodeResult = (data: Data?, error: Error?)
 
 open class CodableHelper {
 
+    open static var dateformatter: DateFormatter?
+
     open class func decode<T>(_ type: T.Type, from data: Data) -> (decodableObj: T?, error: Error?) where T : Decodable {
         var returnedDecodable: T? = nil
         var returnedError: Error? = nil
 
         let decoder = JSONDecoder()
-        decoder.dataDecodingStrategy = .base64
-        if #available(iOS 10.0, *) {
-            decoder.dateDecodingStrategy = .iso8601
+        if let df = self.dateformatter {
+            decoder.dateDecodingStrategy = .formatted(df)
+        } else {
+            decoder.dataDecodingStrategy = .base64
```

Comment:
Why is this inside the `else` scope? 
Shouldn't the `dataDecodingStrategy` be applied, regardless of the custom date formatter?
Could be a mistake because date and data looks pretty much the same...